Open Letter to Meta (#1495)
xsf/xmpp.org: content/about/open-letter-meta-dma/technical-briefing.md

---
title: "Detailed technical briefing: The Case for XMPP – Why Meta Must Embrace True Messaging Interoperability"
layout: single_open_letter_meta_dma

> ---

Reviewer: This briefing is very much aimed at an audience of the EU and press, rather than Meta, which it's frankly quite brutal about (though not incorrect). Would this be included in the letter as sent to meta?

Author: I think we cannot separate them. The letter is too short for substance on its own, so the briefing goes into detail. Meta has the burden/privilege of being a (two-fold) gatekeeper in this area. The letter and briefing have to be seen in light of *why* the EU Commission has designated them as gatekeeper. The EU Commission created this situation by focusing the debate on market competition (hence _Digital Markets Act_), rather than looking to effects on users, and by restricting the current debate to instant messaging only. Advanced (but perceived as must-have by end-users) features like voice and video calling are due in about 2 years.

Reviewer: Hmmm... OK, I think I can work with this and suggest some edits on the technical briefing that will make it less of a bitter pill to swallow. The voice/video aspects are really interesting, since even if they're not covered by the regulations today they clearly will need to be in the future.

Author: From the DMA, article 7:

```
2.   The gatekeeper shall make at least the following basic
     functionalities referred to in paragraph 1 interoperable where
     the gatekeeper itself provides those functionalities to its own
     end users:

(a) following the listing in the designation decision pursuant to Article 3(9):
    (i) end-to-end text messaging between two individual end users;
    (ii) sharing of images, voice messages, videos and other attached
    files in end to end communication between two individual end
    users;

(b) within 2 years from the designation:
    (i) end-to-end text messaging within groups of individual end
    users;
   (ii) sharing of images, voice messages, videos and other attached
   files in end-to-end communication between a group chat and an
   individual end user;

(c) within 4 years from the designation:
   (i) end-to-end voice calls between two individual end users;
   (ii) end-to-end video calls between two individual end users;
   (iii) end-to-end voice calls between a group chat and an individual
        end user;
   (iv) end-to-end video calls between a group chat and an individual
        end user.
```

With the designation on September 6 2023, the respective dates are a) March 6 2024, b) September 6 2025, and c) September 2027.

Reviewer: OK, so I'd like to include something somewhere along the lines of:

XMPP not only provides a well-deployed and understood plan for the immediate demands of one-to-one messaging, but a long term path that includes both groupchat and voice/video calls. Just as it would make little sense for Meta to demand anything but WebRTC as the basis for voice/video, we feel it makes little sense demanding anything but XMPP.

## Introduction

The European **[Digital Markets Act (DMA)](https://digital-markets-act.ec.europa.eu/about-dma_en)** was introduced to ensure fair competition and prevent large technology companies from exploiting their dominant positions. A key aspect of the DMA is the requirement for **Number-Independent Interpersonal Communication Services (NIICS)**—which includes messaging platforms like WhatsApp and Messenger—to provide true interoperability with third-party services. This requirement is essential to breaking down walled gardens, promoting user choice, and fostering a more competitive digital ecosystem.

> The **[XMPP Standards Foundation](https://xmpp.org/about/xmpp-standards-foundation/)** has been closely following the implementation of the Digital Markets Act (DMA) in the European Union, with a specific focus on **messaging interoperability**. As an organization dedicated to open messaging standards, we are committed to helping stakeholders navigate the principles and technical realities of interoperable messaging.

Author: I would add in the next paragraph (in italic):

> For over 25 years, we have developed and refined the **eXtensible Messaging and Presence Protocol (XMPP)**—a robust, extensible, and open standard that has fostered messaging interoperability worldwide. _The XSF, through its membership requirements, and XMPP as a whole, by way of the distributed extensibility of the protocol, are independent of any vendor._ [..]

> The **[XMPP Standards Foundation](https://xmpp.org/about/xmpp-standards-foundation/)** has been closely following the implementation of the Digital Markets Act (DMA) in the European Union, with a specific focus on **messaging interoperability**. As an organization dedicated to open messaging standards, we are committed to helping stakeholders navigate the principles and technical realities of interoperable messaging.

Reviewer: That's perfect.

> The **[XMPP Standards Foundation](https://xmpp.org/about/xmpp-standards-foundation/)** has been closely following the implementation of the Digital Markets Act (DMA) in the European Union, with a specific focus on **messaging interoperability**. As an organization dedicated to open messaging standards, we are committed to helping stakeholders navigate the principles and technical realities of interoperable messaging.

Author: Added in [3cfa1bb](https://github.com/xsf/xmpp.org/pull/1495/commits/3cfa1bb4f8cddd2c1ae0452840497be6064359de)

## Smartphones, WhatsApp, and the Gatekeeping Strategy

Despite XMPP’s success, technical advantages, and suitability for messaging interoperability, its adoption slowed as the messaging landscape evolved. The shift toward proprietary platforms coincided with the rise of smartphones, which led to growing financial stakes in messaging services and increased efforts by dominant players to consolidate control over their networks.

### The Shift to Mobile and the Rise of Proprietary Networks

In **2007**, Apple launched the iPhone, ushering in the smartphone era and the widespread adoption of mobile internet. By **2009**, the introduction of both the App Store and push notification services paved the way for the launch of **WhatsApp**. Messaging, which had previously been limited to SMS on mobile, quickly became a **rich and ubiquitous communication tool**.

> WhatsApp took a strategic approach by leveraging a public resource—the phone number system—as a unique user identifier. Combined with easy access to users’ phone contacts in modern smartphone operating systems, WhatsApp was able to build an extensive **proprietary** messaging network. In essence, it **became the new ICQ**, but at an even larger scale. As the service grew, it positioned itself **in direct competition with mobile carriers**, as **over-the-top (OTT) messaging**—services that bypass traditional carrier networks by leveraging the open internet. At the time, **net neutrality** was seen as a safeguard ensuring that carriers could not abuse their control over communication infrastructure, but it also allowed closed proprietary messaging services to thrive.

Reviewer: I think you want to mention here that WhatsApp started as an XMPP service with a locked-in client. (Although it is mentioned later).

Author: I added a clause in [cd46751](https://github.com/xsf/xmpp.org/pull/1495/commits/cd46751edf74f61167cbaf2dbf3f523f85db558a).

- **Facebook Messenger**
- **WhatsApp**
- **Instagram private messaging**
- **Telegram**
- **WeChat**
- **LinkedIn Messaging**
- **Various Google messaging services**
- And many more

Each of these services **operates in isolation**, forcing users to either create accounts on multiple platforms to stay in touch with all their contacts or rely on a single provider—driving further concentration in the sector. If a user chooses to focus on just one app, they must convince all their contacts to do the same, effectively reinforcing a **global-scale monopoly** in communication.

### Beyond Messaging: The Broader Lock-In of Communication

What is worst, is that this issue is no longer limited to text-based messaging. Modern messengers have expanded to **voice and video calls**, effectively **controlling digital communication as a whole**. If we do not change course, the endgame is clear, as we are stuck in a race toward a **global monopoly on internet communication**, where a handful of gatekeepers could dictate their terms, limiting competition and user freedom.

## The European Commission and the Digital Markets Act

> To prevent extreme concentration of power in the messaging sector, the **European Commission introduced the Digital Markets Act (DMA)**. Its goal is to ensure that communication remains **open, fair, and competitive**, preventing a future where global messaging is controlled by a **monopoly, duopoly, or, at best, an oligopoly**. Enforcing interoperability is the only way to allow users to **freely choose their preferred messaging provider** rather than being locked into dominant platforms.

Reviewer 2: Slightly tweaked the wording - the DMA is not *only* about messaging.

```suggestion
To prevent extreme concentration of power in the messaging sector, the **European Commission included it firmly within the scope of the Digital Markets Act (DMA)**. Its goal is to ensure that communication remains **open, fair, and competitive**, preventing a future where global messaging is controlled by a **monopoly, duopoly, or, at best, an oligopoly**. Enforcing interoperability is the only way to allow users to **freely choose their preferred messaging provider** rather than being locked into dominant platforms.
```

### Key Concerns with Meta’s Interoperability Proposal

#### 1. Restrictive Access Through NDAs and Agreements

Meta requires companies to **sign a non-disclosure agreement (NDA)** to access their interoperability detailed documents and likely demands an agreement before enabling interoperability between networks. This effectively **keeps companies in a gatekeeper relationship**, just as Meta did when launching *Messenger for Businesses* to control interactions and ensure benefits flowed primarily to them. 

For competitors, **signing an agreement with Meta grants them disproportionate leverage** due to their massive user base. This is **not true interoperability**—it is a **negotiated privilege**, reinforcing Meta’s dominant position.

Requiring an NDA **prevents non-profit organizations like the XSF** from evaluating or contributing to the interoperability mechanism. It **blocks open-source developers from implementing interoperability** in their software, severely limiting innovation.

#### 2. API-Based Approach Instead of Open Protocols

Meta’s proposal appears to be based on **APIs rather than open protocols**. This is a fundamental problem because:

> - **Protocols** are designed for **interop and cooperation**, ensuring long-term stability.

Reviewer 2: ```suggestion
- **Protocols** are designed for **interoperability and cooperation**, ensuring long-term stability.
```

> - **APIs** are controlled by the provider and offer a **limited set of features**, with associated **rate limits, fees, and restrictions** that can change over time.

Reviewer 2: Rate limits are a fundamental abuse prevention technique at all levels of the stack. Proposing "usage limits" for clarity in the context of APIs.

```suggestion
- **APIs** are controlled by the provider and offer a **limited set of features**, with associated **usage limits, fees, and restrictions** that can change over time.
```

### Understanding federation: A Telephone Analogy: Why Meta’s Approach is Flawed**

This is why the **XMPP Standards Foundation (XSF) steps in** and propose to enable interoperability on federation principles which are at the hearth of the Internet majors protocols and allowed it to stay open.

To better understand the shortcomings of Meta’s proposal, we can compare it to another essential communication system: **the global phone network**.

Today, phone networks are **interoperable and federated**. No matter which provider you use, you can call anyone on another network—whether it's a landline or mobile service. **You do not need an account with every provider to communicate.** This seamless federation is possible because:

1. **A standardized common protocol** ensures universal communication.
2. **A distributed ID attribution system** allows calls to be routed correctly between providers.

The **same model can and should be applied to messaging interoperability**. 

> The XMPP Standards Foundation has already done the heavy lifting: by **adopting the XMPP S2S protocol as the interoperability standard**, we can **achieve the same seamless communication in messaging**.

Reviewer 2: This is a technical document, but this abbreviation may not be familiar to readers outside the community.

```suggestion
The XMPP Standards Foundation has already done the heavy lifting: by **adopting the XMPP S2S (server-to-server) protocol as the interoperability standard**, we can **achieve the same seamless communication in messaging**.
```

If Meta has specific requirements, such as:

- Features unique to Facebook Messenger or WhatsApp
- Scalability considerations due to the size of Meta’s user base  
- New extensions to address Meta’s specific use cases  

The XSF is the right place to have these discussions. Our mission is to evolve and improve federated messaging standards.

### Join the Conversation on the Future of Federated Messaging

We understand that Meta may wish to contribute to the development of XMPP, and we would welcome Meta’s most prominent messaging experts to bring their expertise, concerns, and ideas into the work of the XMPP Standards Foundation.

Let’s work together to build a truly interoperable, open, and competitive messaging future.

***

### Contact Information

Press and general inquiries: [info@xmpp.org](mailto:info@xmpp.org)

Ralph Meijer  
[Chair of the Board](/about/xsf/people#chair)  
XMPP: [ralphm@ik.nu](xmpp:ralphm@ik.nu)  
Mail: [ralphm@ik.nu](mailto:ralphm@ik.nu)

Matthew Wild  
[Executive Director](/about/xsf/people#executive-director)  
XMPP: [me@matthewwild.co.uk](xmpp:me@matthewwild.co.uk)  
Mail: [me@matthewwild.co.uk](mailto:me@matthewwild.co.uk)

Mickaël Rémond  

> Member and Author  

Reviewer: We should give Mickael a special title. Liaison or something.

Author: In be6b0a53.
